x/ibc/04-channel/keeper/packet_test.go

```diff
@@ -4,92 +4,157 @@ import (
 	"fmt"
 
 	capabilitytypes "github.com/cosmos/cosmos-sdk/x/capability/types"
-	transfertypes "github.com/cosmos/cosmos-sdk/x/ibc-transfer/types"
-	connection "github.com/cosmos/cosmos-sdk/x/ibc/03-connection"
+	clientexported "github.com/cosmos/cosmos-sdk/x/ibc/02-client/exported"
+	connectiontypes "github.com/cosmos/cosmos-sdk/x/ibc/03-connection/types"
 	"github.com/cosmos/cosmos-sdk/x/ibc/04-channel/exported"
 	"github.com/cosmos/cosmos-sdk/x/ibc/04-channel/types"
+	commitmenttypes "github.com/cosmos/cosmos-sdk/x/ibc/23-commitment/types"
 	host "github.com/cosmos/cosmos-sdk/x/ibc/24-host"
+	ibctesting "github.com/cosmos/cosmos-sdk/x/ibc/testing"
 )
 
+var (
+	validPacketData          = []byte("VALID PACKET DATA")
+	disabledTimeoutTimestamp = uint64(0)
+	disabledTimeoutHeight    = uint64(0)
+	timeoutHeight            = uint64(100)
+
+	// for when the testing package cannot be used
+	clientIDA  = "clientA"
+	clientIDB  = "clientB"
+	connIDA    = "connA"
+	connIDB    = "connB"
+	portID     = "portid"
+	channelIDA = "channelidA"
+	channelIDB = "channelidB"
+)
+
+// TestSendPacket tests SendPacket on chainA
 func (suite *KeeperTestSuite) TestSendPacket() {
```

Comment:
Can we test that sending a packet out of order on an ORDERED channel fails?

Reply:
and that sending a packet out of order on an UNORDERED channel succeeds?

Reply:
> and that sending a packet out of order on an UNORDERED channel succeeds?

that is incorrect behaviour. All channels must `send` in order, UNORDERED channels may receive out of order

Reply:
from ics spec: ` abortTransactionUnless(packet.sequence === nextSequenceSend)`

Reply:
Correct, my bad